openapi-cli/src/test/java/io/ballerina/openapi/cmd/OpenApiGenServiceCmdTest.java

```diff
@@ -111,4 +113,34 @@ public void testOneOfSchemaGen() throws IOException {
             Assert.fail("Service generation for OneOf Schema type failed.");
         }
     }
+
+    @Test(description = "Test for --without-data-binding flag")
+    public void testWithoutDataBinding() throws IOException {
+        Path yamlPath = resourceDir.resolve(Paths.get("withoutDataBinding.yaml"));
+        String[] args = {"--input", yamlPath.toString(), "--without-data-binding", "-o",
+                this.tmpDir.toString(), "--mode", "service"};
+        OpenApiCmd cmd = new OpenApiCmd(printStream, this.tmpDir);
+        new CommandLine(cmd).parseArgs(args);
+        String output = "";
+        try {
+            cmd.execute();
+        } catch (BLauncherException e) {
+            output = e.getDetailedMessages().get(0);
+        }
+
+        Path expectedServiceFile = resourceDir.resolve(Paths.get("expected_gen",
+                "without-data-binding.bal"));
+        Stream<String> expectedServiceLines = Files.lines(expectedServiceFile);
+        String expectedService = expectedServiceLines.collect(Collectors.joining("\n"));
+        expectedServiceLines.close();
+        Assert.assertFalse(Files.exists(this.tmpDir.resolve("types.bal")));
+        if (Files.exists(this.tmpDir.resolve("withoutdatabinding_service.bal"))) {
+            String generatedService = getStringFromFile(this.tmpDir.resolve("withoutdatabinding_service.bal"));
+            Assert.assertEquals(replaceWhiteSpace(generatedService), replaceWhiteSpace(expectedService),
+                    "Expected content and actual generated content is mismatched for: " + yamlPath);
+            deleteGeneratedFiles("without-data-binding-service.bal");
+        } else {
+            Assert.fail("Service generation for All Of Schema type failed.");
```

Comment:
is this error message correct?
```suggestion
            Assert.fail("Service generation for All Of Schema type failed.");
```

Reply:
This is a mistake and updated according to the test.